File: src/main/java/com/fullstack/demo/entity/Post.java
package com.fullstack.demo.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.Date;
import java.util.HashSet;
import java.util.Set;

@Entity
@Getter
@Setter
@Table(name = "Post")
public class Post {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long postId;

    @Column(nullable = false, length = 50)
    private String postTitle;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String postContent;

    @Column(length = 255)
    private String thumbnailImageUrl;

    @Column(length = 255)
    private String contentImageUrl;

    @Column
    private Date startDate;

    @Column
    private Date endDate;

    @Column(nullable = false)
    private LocalDateTime createdAt;

    @Column(nullable = false)
    private LocalDateTime updatedAt;

    @Column
    private Integer viewCount;

    @Column
    private Integer likeCount;

    @Column(length = 255)
    private String postCategory;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "board_id", nullable = false)
    private Board board;  // Board와의 관계 설정


    @ManyToMany
    @JoinTable(
            name = "PostCategory",
            joinColumns = @JoinColumn(name = "post_id"),
            inverseJoinColumns = @JoinColumn(name = "category_id")
    )
    private Set<Category> categories = new HashSet<>();


    // Getters and setters

    public void setBoard(Board board) {
        this.board = board;
    }

    // Getters and setters
}

File: lib/entity/Post.dart
class Post {
  final int postId;
  final String postTitle;
  final String postContent;
  final String? thumbnailImageUrl;
  final String? contentImageUrl;
  final DateTime? startDate;
  final DateTime? endDate;
  final DateTime createdAt;
  final DateTime updatedAt;
  final int viewCount;
  final int likeCount;
  final String? postCategory;

  Post({
    required this.postId,
    required this.postTitle,
    required this.postContent,
    this.thumbnailImageUrl,
    this.contentImageUrl,
    this.startDate,
    this.endDate,
    required this.createdAt,
    required this.updatedAt,
    required this.viewCount,
    required this.likeCount,
    this.postCategory,
  });

  factory Post.fromJson(Map<String, dynamic> json) {
    return Post(
      postId: json['postId'],
      postTitle: json['postTitle'],
      postContent: json['postContent'],
      thumbnailImageUrl: json['thumbnailImageUrl'],
      contentImageUrl: json['contentImageUrl'],
      startDate: json['startDate'] != null ? DateTime.parse(json['startDate']) : null,
      endDate: json['endDate'] != null ? DateTime.parse(json['endDate']) : null,
      createdAt: DateTime.parse(json['createdAt']),
      updatedAt: DateTime.parse(json['updatedAt']),
      viewCount: json['viewCount'] ?? 0,
      likeCount: json['likeCount'] ?? 0,
      postCategory: json['postCategory'],
    );
  }
}